spark/src/main/scala/geotrellis/spark/clip/ClipToGrid.scala

```diff
@@ -132,73 +140,53 @@ object ClipToGrid {
         def coveredBy(e: Extent) = pg.coveredBy(e.toPolygon.jtsGeom)
       }
 
-    def gcPredicates =
+    val gcPredicates =
       new Predicates {
         def covers(e: Extent) = feature.geom.jtsGeom.covers(e.toPolygon.jtsGeom)
         def coveredBy(e: Extent) = feature.geom.jtsGeom.coveredBy(e.toPolygon.jtsGeom)
       }
 
-    def clipToKey(
-      k: SpatialKey,
-      preds: Predicates
-    ): Option[(SpatialKey, Feature[Geometry, D])] = {
-      val extent = mapTransform(k)
-
-      clipFeature(extent, feature, preds).map(k -> _)
-    }
-
     val iterator: Iterator[(SpatialKey, Feature[Geometry, D])] =
       feature.geom match {
-        case p: Point =>
-          val k = mapTransform(p)
-          clipToKey(k, pointPredicates).toSeq.iterator
+        case p: Point => Iterator.single(mapTransform(p) -> feature)
         case mp: MultiPoint =>
           mp.points
             .map(mapTransform(_))
-            .distinct
-            .map(clipToKey(_, mpOrLinePredicates))
-            .flatten
+            .toSet
```

Comment:
`toSet` is always faster than `distinct` for collections smaller than ~100k elements.